src/main/java/com/microsoft/graph/concurrency/ChunkedUploadResponseHandler.java

```diff
@@ -154,30 +159,35 @@ public void configConnection(final Response response) {
             final ILogger logger) throws Exception {
         InputStream in = null;
         try {
-        	if (response.code() == HttpResponseCode.HTTP_ACCEPTED) {
-        		logger.logDebug("Chunk bytes has been accepted by the server.");
-        		in = new BufferedInputStream(response.body().byteStream());
-        		final UploadSession session = serializer.deserializeObject(
-        				DefaultHttpProvider.streamToString(in), UploadSession.class);
-
-        		return new ChunkedUploadResult<UploadType>(session);
-
-        	} else if (response.code() == HttpResponseCode.HTTP_CREATED
-        			|| response.code() == HttpResponseCode.HTTP_OK) {
-        		logger.logDebug("Upload session is completed, uploaded item returned.");
-        		in = new BufferedInputStream(response.body().byteStream());
-        		String rawJson = DefaultHttpProvider.streamToString(in);
-        		UploadType uploadedItem = serializer.deserializeObject(rawJson,
-        				this.deserializeTypeClass);
-
-        		return new ChunkedUploadResult<UploadType>(uploadedItem);
-        	} else if (response.code() >= HttpResponseCode.HTTP_CLIENT_ERROR) {
+        	if (response.code() >= HttpResponseCode.HTTP_CLIENT_ERROR) {
         		logger.logDebug("Receiving error during upload, see detail on result error");
 
         		return new ChunkedUploadResult<UploadType>(
         				GraphServiceException.createFromConnection(request, null, serializer,
         						response, logger));
-        	}
+            } else if (response.code() >= HttpResponseCode.HTTP_OK 
+                    && response.code() < HttpResponseCode.HTTP_MULTIPLE_CHOICES) {
+                final Map<String, String> headers = responseHeadersHelper.getResponseHeadersAsMapStringString(response);
+                final String contentType = headers.get(Constants.CONTENT_TYPE_HEADER_NAME);
+                final String location = headers.get("Location");
+                if(contentType != null
```

Comment:
If contenttype and location are null, we should have an `else` where we log this unexpected scenario at the debug level.

Reply:
added, thanks